common/lib/transport/connectionmanager.js

```diff
@@ -639,8 +650,9 @@ var ConnectionManager = (function() {
 				});
 				return;
 			}
-			/* FIXME: decide if fatal */
-			var fatal = false;
+			/* all codes in the 40-50k zone indicate a fault with the request, so
```

Comment:
Are you sure this is what's specified? A 502 for example would suggest that a fallback host should be attempted.